lib/ansible/module_utils/fortios.py

```diff
@@ -42,6 +44,14 @@
 except ImportError:
     HAS_PYFG = False
 
+try:
+    import requests
+    from requests import ConnectionError
+
+    HAS_REQUESTS = True
+except ImportError:
+    HAS_REQUESTS = False
```

Comment:
We have previously discussed that module are not permitted to uses the requests module, due to unnecessarily adding dependencies to modules.  You should utilize the functionality in `ansible.module_utils.urls` instead of requests.

Reply:
One of the reasons for using requests proxy was that it easily supports socks5 proxy, which is a requirement in our environment. It seems that the urls module doesn't support it now ("Currently ansible only supports HTTP proxies"), so unfortunately dependency on requests (or any other external library to handle HTTP connections) is required.

Reply:
That functionality should be added to `urls.py`

Reply:
I am afraid that my limited coding capabilities will not allow me to add socks proxy support to the urls.py module in a reasonable timeframe. Is lack of dependency on requests a hard requirement for the module to be accepted by Ansible?

Reply:
Is the dependency on a socks proxy part of your specific environment, or something that every person using this module will need?

I have more reviewing to do, but I would be fundamentally against the module having specific settings for socks if it isn’t a requirement of fortios.

Just as with using other proxies it should be a fundamental part of how Ansible can be used and should not be a determining factor of how a module is implemented due to one persons environmental restrictions.

Reply:
The dependency on socks proxy is requirement of my specific environment (to be specific, it's some of the devices in my environment, others don't need any proxy to connect to them), but it is important to note that my employer is paying me salary for managing this specific environment.
My employer is understanding the importance of open source and is ok with me spending extra time to make this module part of ansible (especially since working on documentation and tests is making the code better), but I am afraid that adding the code to urls.py and then changing fortios.py to achieve exactly what the module is already doing successfully would be a stretch for my employer. And unfortunately due to my relative weakness of coding and personal reasons, I will not be able to devote sufficient personal time to solve this issue in the near future.
Would it be possible to have module rely on requests library for the time being and then use the power of open source community to make necessary adjustments to urls.py and this module to make it not rely on the  requests library?